Deadlock.cs

```diff
@@ -0,0 +1,45 @@
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading;
+using System.Threading.Tasks;
+
+namespace CodeReviewDemo
+{
+    internal class DeadlockDemo
+    {
+        object myLock1 = new object();
+        object myLock2 = new object();
+        void StartThreads()
+        {
+            Thread t1 = new Thread(ExecuteThread1);
+            Thread t2 = new Thread(ExecuteThread2);
+            t1.Start();
+            t2.Start();
+        }
```

Comment:
Using `Thread.Sleep` to simulate a delay can cause unnecessary performance issues and is not recommended for production code. Consider using proper synchronization mechanisms or task delays.